### PR TITLE
#479. Repo#activate() throws RepoAlreadyActiveException if active.

### PR DESCRIPTION
**Pull request by criske**
PR for #479

### CODE REVIEW

File: self-api/src/main/java/com/selfxdsd/api/exceptions/RepoAlreadyActiveException.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * @since 0.0.20
+ */
+public final class RepoAlreadyActiveException extends RuntimeException {
+
+    /**
+     * Ctor.
+     * @param repo Repo in question.
+     */
+    public RepoAlreadyActiveException(final Repo repo){
+        super("Repo \"" + repo.fullName() + "\" is already active.");
```
Comment (criske) [inline self-api/src/main/java/com/selfxdsd/api/exceptions/RepoAlreadyActiveException.java:18]: I've tried to keep the same pattern as `InvoiceAlreadyPaidException`.
But in this case `repo.fullName()` will require another json request. So unless we find a way to cache these json responses (etags or something) -- or at least part of them like `fullName`, maybe we should pass the `fullName` string instead. (and possibly the `Provider` name, just to make the message clearer.)

Comment (amihaiemil) [reply self-api/src/main/java/com/selfxdsd/api/exceptions/RepoAlreadyActiveException.java:18]: @criske Method [BaseRepo.json()](https://github.com/self-xdsd/self-core/blob/master/self-core-impl/src/main/java/com/selfxdsd/core/BaseRepo.java#L91) caches the json so there will be only one initial real call made, right? 

Or am I missing something? :D

**Review by amihaiemil — COMMENTED** [review]
@criske looks nice, see my reply :D

File: self-api/src/main/java/com/selfxdsd/api/exceptions/RepoAlreadyActiveException.java (hunk 1 above)
Comment (criske) [reply self-api/src/main/java/com/selfxdsd/api/exceptions/RepoAlreadyActiveException.java:18]: You're right, dunno why I had the impression it didn't lol. You can merge it

**Comment by amihaiemil** [comment]
@rultor merge it please

**Pull request merged**